listener.go

```diff
@@ -10,9 +9,11 @@ import (
 
 // Listen creates a DTLS listener
 func Listen(network string, laddr *net.UDPAddr, config *Config) (*Listener, error) {
-	if config == nil {
-		return nil, errors.New("no config provided")
+	err := validateConfig(config)
```

Comment:
Personally I like one liners for this better:

``` go
if err := validateConfig(config); err != nil {
    //…
}
```

Reply:
Mmm yes, since err isn't used any further that's kinda nicer. @hjames9 could you address this one, then I'll merge it.